Misc updates

adamshire123 opened the pull request:
### What does this PR do?

- Updates Readme instructions for creating sample data and local testing
- Adds a check for valid SAP sequence number, and a test for the check.
- bumps dependency versions

### Helpful background context
During testing we learned that SAP sequence numbers must be at least three digits long. 
The production sequence number is in the thousands and shouldn't reset, so this shouldn't be an issue
in production, but it was during testing where our fake sap sequence number was '1'. 

The process to create sample data changed slightly. `README.me` was updated accordingly.

### Includes new or updated dependencies?

YES

### Developer

- [x] All new ENV is documented in README (or there is none)
- [x] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

adamshire123 committed:
check for valid sap sequence number

adamshire123 committed:
update dependencies

ehanson8 commented on sapinvoices/sap.py:
```diff
@@ -573,11 +577,16 @@ def generate_next_sap_sequence_number() -> str:
     sap_sequence_parameter = ssm.get_parameter_value(sap_config["SAP_SEQUENCE_NUM"])
     split_parameter = sap_sequence_parameter.split(",")
+    if len(split_parameter[0]) < 3:
+        raise SapSequenceError("SAP sequence number must be three or more digits.")
```
I would include the value here, e.g. `raise SapSequenceError("Invalid SAP sequence:  '%s', number must be three or more digits.", split_parameter[0])`

ehanson8 commented on tests/test_sap.py:
```diff
@@ -571,6 +571,16 @@ def test_generate_next_sap_sequence_number():
 
 
+@pytest.mark.usefixtures("mocked_ssm_bad_sequence_number")
+def test_generate_next_sap_sequence_number_fail():
```
This works without the pytest decorator and just `def test_generate_next_sap_sequence_number_fail(mocked_ssm_bad_sequence_number):`. That appears to override the autouse `mocked_ssm` fixture

adamshire123 replied:
`@pytest.mark.usefixtures` makes it so that pylint doesn't complain about unused arguments.

ehanson8 replied:
Got it, that seems fine

ehanson8 commented on tests/test_sap.py:
```diff
@@ -571,6 +571,16 @@ def test_generate_next_sap_sequence_number():
 
+@pytest.mark.usefixtures("mocked_ssm_bad_sequence_number")
+def test_generate_next_sap_sequence_number_fail():
+    ssm = SSM()
```
`SSM()` could be a fixture since it's called in several tests. I did something similar in the original `alma-scripts`

ehanson8 reviewed (commented):
A few suggestions

ehanson8 commented on sapinvoices/sap.py:
```diff
@@ -586,7 +586,10 @@ def generate_next_sap_sequence_number() -> str:
     if len(split_parameter[0]) < 3:
-        raise SapSequenceError("SAP sequence number must be three or more digits.")
+        raise SapSequenceError(
+            f"Invalid SAP sequence: '{split_parameter[0]}', number must be three or more"
+            "digits."
```
Add space before "digits"

adamshire123 committed:
update readme and SapSequence error message

- add a space to SapSequenceError message
- add local testing instructions to readme

ehanson8 reviewed (approved):
There's a `llama` reference in `README.md` that should be removed but otherwise looks great!
